proto/src/rr/domain/name.rs

```diff
@@ -952,33 +954,46 @@ impl FromStr for Name {
 /// Conversion into a Name
 pub trait IntoName: Sized {
     /// Convert this into Name
-    fn into_name(self: Self) -> ProtoResult<Name>;
+    fn into_name(&self) -> ProtoResult<Name>;
```

Comment:
Functions named `into_()` should take `self` not `self`. So, you should rename this trait to "ToName" and rename the function "to_name".

Note that the caller can always to `x.clone().into_name()` if they want to retain `x`, but with the change no caller would be able to avoid any copy when it isn't necessary. This is why move semantics (the original way) are generally preferable to copy semantics in Rust.

In other words, in general this kind of change is not ideal.

Reply:
Yes. Good feedback. I’m actually going to revert this.